contracts/oracles/TwapOracle.sol

```diff
@@ -131,8 +146,9 @@ contract TwapOracle is OwnableUpgradeable, OracleInterface {
      * @param vToken vToken address
      * @return price in USD, with 18 decimals
      */
-    function getUnderlyingPrice(address vToken) external override view returns (uint256) {
+    function getUnderlyingPrice(address vToken) external override returns (uint256) {
         require(tokenConfigs[vToken].vToken != address(0), "vToken not exist");
+        updateTwap(vToken);
```

Comment:
If this is no longer a getter and includes a mutation does it need a better name?

Reply:
I don't see an issue keeping the same function name.

Reply:
I would also expect a function called `getX` not to have any side effects

Reply:
Added a function name fetchUnderlyingPrice which updates the state